profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,563 @@
+                    return match;
+                }
+            }).orNull();
+        }
+        if (lan == null) {
+            logger.warn("Could not find an existing lan Creating one....");
+            lan = api.lanApi().create(Lan.Request.creatingBuilder()
+                    .dataCenterId(dataCenterId)
+                    .isPublic(Boolean.TRUE)
+                    .name("lan " + name)
+                    .build());
+        }
+
+        //add a NIC to the server
+        waitDcUntilAvailable.apply(dataCenterId);
+        waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+        int lanId = DEFAULT_LAN_ID;
+        if (options.getNetworks() != null) {
+            try {
+                String networkId = Iterables.get(options.getNetworks(), 0);
+                lanId = Integer.valueOf(networkId);
+            } catch (Exception ex) {
+                logger.warn("no valid network id found from options. using default id='%d'", DEFAULT_LAN_ID);
+            }
+        }
+
+        waitDcUntilAvailable.apply(dataCenterId);
+        waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+
+        Nic nic = api.nicApi().create(Nic.Request.creatingBuilder()
+                .dataCenterId(dataCenterId)
+                .name("jclouds" + name)
+                .dhcp(Boolean.TRUE)
+                .lan(lanId)
+                .firewallActive(Boolean.FALSE)
+                .serverId(serverId).
+                build());
+
+        waitNICUntilAvailable.apply(NicRef.create(dataCenterId, serverId, nic.id()));
+        waitDcUntilAvailable.apply(dataCenterId);
+        waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+
+//        connect the rest of volumes to server;delete if fails
+        final int volumeCount = volumeIds.size();
+        for (int j = 1; j < volumeCount; j++) { // skip first; already connected
+            final String volumeId = volumeIds.get(j);
+            try {
+                logger.trace("<< connecting volume '%s' to server '%s'", volumeId, serverId);
+                provisioningManager.provision(jobFactory.create(group, new Supplier<Object>() {
+
+                    @Override
+                    public Object get() {
+                        return api.serverApi().attachVolume(
+                                Server.Request.attachVolumeBuilder()
+                                .dataCenterId(dataCenterId)
+                                .serverId(serverId)
+                                .volumeId(volumeId)
+                                .build()
+                        );
+                    }
+                }));
+
+                logger.trace(">> volume connected.");
+            } catch (Exception ex) {
+                // delete unconnected volume
+                logger.warn(ex, ">> failed to connect volume '%s'. deleting..", volumeId);
+                destroyVolume(volumeId, dataCenterId);
+            }
```

Comment:
Prior this volume-creating code block, a [*bootVolume*](https://github.com/jclouds/jclouds-labs/pull/292/files#diff-3cfe9775f2121ca815ae0fce43f9d211R187) is attached to the server. That server-volume pair already constitutes a jcloud node. The additional volumes here are more like, additional storages, not necessarily making the node unusable if they fail in attaching.